beacon-chain/forkchoice/doubly-linked-tree/types.go

```diff
@@ -11,11 +11,12 @@ import (
 
 // ForkChoice defines the overall fork choice store which includes all block nodes, validator's latest votes and balances.
 type ForkChoice struct {
-	store          *Store
-	votes          []Vote // tracks individual validator's last vote.
-	votesLock      sync.RWMutex
-	balances       []uint64                    // tracks individual validator's last justified balances.
-	balancesByRoot forkchoice.BalancesByRooter // handler to obtain balances for the state with a given root
+	store             *Store
+	votes             []Vote // tracks individual validator's last vote.
+	votesLock         sync.RWMutex
+	balances          []uint64                    // tracks individual validator's balances last accounted in votes.
+	justifiedBalances []uint64                    // tracks individual validator's last justified balances.
```

Comment:
Why do we need to track `balances` anymore?

Reply:
Because we need the old balances to process attestations. To get rid of this we would need to process attestations immediately after updating justified checkpoint. It's an option, but it may be dangerous and it may screw up during init sync